### DISCUSSION TITLE
What is Tasmota Arduino Core? Where should i pay attention to this?

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#20069] [Q&A]
i am not firm with Arduino Core Versions. It is enought to take the latest (dev) repo? is that a preperation that has to be done in plattform.io? Where ist that specifed?

**Comment by sfromis** [comment]
In most situations you do not need to worry, as the Tasmota setup already refers to the relevant Arduino Core version. Deviation from that is mostly for testing a different Core version.

A current case for selecting a different Core version is with Arduino Core 3.0, where there are build environments referring to this
https://github.com/arendst/Tasmota/blob/development/platformio_tasmota_core3_env_sample.ini
https://github.com/arendst/Tasmota/blob/development/platformio_tasmota32.ini#L102-L106
That is if you want to test how well it works with this instead of the regular supported one.

Instead of doing all the hardware handling low level, Tasmota is using Arduino Core which has a lot of features for such things, making the coding needed in Tasmota much less. However, Tasmota is using a build of Arduino Core adapted for Tasmota.

For ESP8266, it is unlikely that there will be further versions of Arduino Core, the above is for ESP32.

**Comment by Jason2866** [comment]
Short answer, you don't have to think about. All setup is already done.
